app/src/main/java/com/example/nigeriatelemedicineapp/registerpatient/RegisterPatientViewModel.kt

```diff
@@ -10,100 +11,130 @@ import retrofit2.Call
+        } else if (!StringUtils.check(phone)) {
+            status.value = Status.EMPTY_PHONE
+        }
+        else{
+            status.value= Status.SUCCESS
+            registerPatient(firstname,lastname,dob,gender,phone)
+        }
+    }
+
+    fun registerPatient(firstname: String, lastname: String, dob: String, gender: String, phone: String) {
+            getIdentifier(firstname, lastname, dob, gender, phone)
     }
 
-    private fun getIdentifier(firstname : String, lastname: String,dob : String,gender : String,phone : String) {
+    private fun getIdentifier(firstname: String, lastname: String, dob: String, gender: String, phone: String) {
         repository.getIdentifier()?.enqueue(object : Callback<IdentifierList> {
             override fun onResponse(call: Call<IdentifierList>, response: Response<IdentifierList>) {
                 if (response.code() == 200) {
-                    val Uuid=response.body()?.getUuid()
-                    identifier.postValue(Uuid)
-                    createPatient(Uuid,firstname,lastname,dob,gender,phone)
+                    val Uuid = response.body()?.getUuid()
+                    createPatient(Uuid, firstname, lastname, dob, gender, phone)
                 }
             }
+
             override fun onFailure(call: Call<IdentifierList>, t: Throwable) {
-                    identifier.postValue("Error fetching Identifier")
+                dialog.postValue(false)
             }
         })
     }
 
-    private fun createPatient(Uuid: String?,firstname : String, lastname: String,dob : String,gender : String,phone : String) {
+    private fun createPatient(
+        Uuid: String?,
+        firstname: String,
+        lastname: String,
+        dob: String,
+        gender: String,
+        phone: String
+    )
+    {
         val name = Name()
-        name.givenName=firstname
-        name.familyName=lastname
-
-        val attribute= Attribute()
-        attribute.attributeType= AttributeType()
-        attribute.value=phone
-
-        val person= Person()
-        person.names= listOf(name)
-        person.gender=gender
-        person.birthdate=dob
-        person.attributes= listOf(attribute)
-
-        val identifier= Identifier()
-        identifier.identifier=Uuid
-
-        Timber.d("Creating patient with \n" +
-                "${identifier.identifier} \n" +
-                "and name ${name.givenName} ${name.familyName} \n" +
-                "DOB : $dob \n" +
-                "gender : $gender    \n" +
-                "phone : $phone")
-
-        val patient= Patient()
-        patient.identifiers= listOf(identifier)
-        patient.person=person
+        name.givenName = firstname
+        name.familyName = lastname
+
+        val attribute = Attribute()
+        attribute.attributeType = AttributeType()
+        attribute.value = phone
+
+        val person = Person()
+        person.names = listOf(name)
+        person.gender = gender
+        person.birthdate = dob
```

Comment:
Don't we have to do some kind of conversion here between the date format common in Nigeria and the format accepted by OpenMRS? Or has that not been implemented yet?